internal/domain/service/auth_service.go

```
@@ -35,24 +40,14 @@ func (a *authService) Authenticate(
-	authProviderName, err := oauth.NewAuthProviderName(providerName)
-	if err != nil {
-		utils.HandleError(ctx, err, "NewAuthProviderName")
-		return nil, errtrace.Wrap(messages.InvalidProviderError)
-	}
-
-	provider, err := oauth.NewOIDCProvider(ctx, authProviderName, a.config)
-	if err != nil {
-		utils.HandleError(ctx, err, "OIDCProviderFactory")
-		return nil, errtrace.Wrap(err)
-	}
+	provider, err := a.authProviderFactory.NewAuthProvider(ctx, providerName)
```

golangci-lint (1.62.2)

ineffectual assignment to err

(ineffassign)